### ISSUE TITLE
Regression - Incorrect type for JsonArray on JwtPayload

### ISSUE THREAD

**Issue by richard-jfc** [aws-amplify/amplify-js#13667]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Not applicable

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

None

### Environment information

N/A

### Describe the bug

The [typing on JsonArray](https://github.com/aws-amplify/amplify-js/blob/6daf0d381dd05f0d4ddb22c69580c053d94c7491/packages/core/src/singleton/Auth/types.ts#L23) appears to be incorrect since: https://github.com/aws-amplify/amplify-js/pull/13010

It currently does not allow Arrays of objects.

Eg:
```json
{
  "abc": [
    {
      "an": "object"
    }
  ]
}
```

Which is valid JSON.

The [identities](https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-the-id-token.html#user-pool-id-token-payload) field on the payload of a cognito JWT includes a JSON array of objects, which do not meet this typing.

### Expected behavior

The Typing on JsonArray should be reverted/updated to allow objects within arrays.

### Reproduction steps

1. Install the latest `aws-amplify/auth`
2. Retrieve a JWT payload using: 
```typescript
import {  fetchAuthSession } from 'aws-amplify/auth';
const session = await fetchAuthSession();
const { accessToken, idToken } = session.tokens;

// This will fail because it doesn't meet the JsonArray type
idToken!.payload!.identities as {
  userId: string;
  providerName: string;
  providerType: string;
  primary: string;
  dateCreated: string;
  issuer: string;
}[]

```
3. Attempt to retrieve the identities field which should be:
```typescript
{
  userId: string;
  providerName: string;
  providerType: string;
  primary: string;
  dateCreated: string;
  issuer: string;
}[]
```
This does not meet the type of JsonArray.

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by cwomack** [comment]
Hello, @richard-jfc 👋.  This is a great catch and we will mark this as a bug and update the types to include an array of JSON objects.  If you'd like, we always encourage our community to open a PR for the suggested change, but will review this internally either way.  Thank you for opening this issue!

**Comment by cwomack** [comment]
Following up here, @richard-jfc 👋.   The fix for this issue (PR #13669) was released in v6.5.0 of Amplify.  Can you see if upgrading to the most recent version helps resolve this for you?

**Comment by richard-jfc** [comment]
Yep that's working, that was quick, thanks!
